### PR TITLE
gpu: Migrate implementation to the cpp file

### PR DESCRIPTION
**Pull request by ameerj**
This PR mainly aims to reduce the amount of dependencies in `video_core/gpu.h`, as it is a very widely included header file, in an effort to reduce compiling overhead. The definition and implementation of `Tegra::GPU` has moved to the cpp file, with the header now serving only as an interface to its public methods. As a result, many unusued headers were trimmed.

Through this migration, two other changes were needed: the `GPU::LockSync` method's single usage was refactored, and the VSync thread was converted to a `std::jthread` to avoid a deadlock on stopping emulation. Further testing is needed to ensure this refactoring did not introduce other anomalies.

### CODE REVIEW

File: src/video_core/texture_cache/texture_cache_base.h [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 #pragma once
 
-#include <array>
 #include <mutex>
 #include <span>
 #include <type_traits>
 #include <unordered_map>
 #include <unordered_set>
 #include <vector>
+#include <queue>
```
Comment (lioncash) [inline src/video_core/texture_cache/texture_cache_base.h:13]: Shouldn't this be below `<mutex>`?

Comment (ameerj) [reply src/video_core/texture_cache/texture_cache_base.h:12]: clang format says no 🤷

Comment (Morph1984) [reply src/video_core/texture_cache/texture_cache_base.h:12]: can confirm, clang format somehow places <queue> at the bottom

Comment (FernandoS27) [reply src/video_core/texture_cache/texture_cache_base.h:12]: I'll just accept this, no idea why clang format is wrong there.

**Review by FernandoS27 — APPROVED** [review]

**Pull request merged**